src/query/case.rs

```diff
@@ -0,0 +1,113 @@
+use crate::{Condition, Expr, SimpleExpr};
+
+#[derive(Debug, Clone)]
+pub(crate) struct CaseStatementCondition {
+    /// The condition in the `WHEN` clause of a case statement.
+    /// If None, query builder will infer this to be the closing
+    /// `ELSE` clause. Note that `ELSE` is optional and will return NULL
+    /// if `ELSE` is not present but the condition is met.
+    pub(crate) condition: Option<Condition>,
+    /// The result if the condition is met. This is the `THEN` clause.
+    pub(crate) result: Expr,
+}
+
+#[derive(Debug, Clone)]
+pub struct CaseStatement {
+    pub(crate) conditions: Vec<CaseStatementCondition>,
```

Comment:
I think better struct is:
```
struct CaseStatement {
    conditions: Vec<...>,
    else: Option<...>,
}
```
Because grammar is:
```
CASE WHEN condition THEN result
     [WHEN ...]
     [ELSE result]
END
```
and `WHEN` `ELSE` are different parts